v63002/inref/gtm5250.m

```diff
@@ -0,0 +1,47 @@
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;								;
+; Copyright (c) 2018 YottaDB LLC. and/or its subsidiaries.	;
+; All rights reserved.						;
+;								;
+;	This source code contains the intellectual property	;
+;	of its copyright holder(s), and is made available	;
+;	under a license.  If you do not know the terms of	;
+;	the license, please stop and do not read further.	;
+;								;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;
+tpnotacid
+	tstart ():(serial:transaction="BA")
+	if $trestart>2 hang .999
+	else  do
+	. set ^Y=$increment(^i)
+	. zsystem "$ydb_dist/mumps -run ^%XCMD 'set ^Y=$increment(^i)'"
+	tcommit
+	quit
+
+locktimeout
+	set jmaxwait=0
+	set ^X(1)=0
+	set ^X(2)=0
+	do ^job("lockchild^gtm5250",1,"""""")
+	for  quit:^X(1)  hang 1
+	lock ^A(1):1.234
```

Comment:
I would like a test that a lock with timeout of 1.234 seconds times out somewhere between 1.23 and 1.24 seconds. That would be a test that fractional timeouts are really supported. One might not see that happening right away due to system load issues so one might need to keep trying this again say a max of 10 times before issuing an error. Same with OPEN command.

Reply:
Forgot to mention. You can use $ZUT before and after the command and measure the difference and compare that value (in microseconds) against the timeout. Also, please randomly choose 1.2345 or 1.23456 instead of 1.234 to confirm that even if a precision more than millisecond is specified, it is rounded to the nearest millisecond.